src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -420,6 +420,33 @@ public static async Task WaitForReplicationAsync(DocumentDatabase database, Time
             }
         }
 
+        private static void ThrowTimeoutException(List<WaitForIndexItem> indexesToWait, int i, Stopwatch sp, QueryOperationContext context, long cutoffEtag)
+        {
+            var staleIndexesCount = 0;
+            var erroredIndexes = new List<string>();
+
+            for (var j = i; j < indexesToWait.Count; j++)
+            {
+                var index = indexesToWait[j].Index;
+                
+                if (index.State == IndexState.Error)
+                    erroredIndexes.Add(index.Name);
+                
+                if (index.IsStale(context, cutoffEtag))
+                    staleIndexesCount++;
+            }
+
+            var errorMessage = $"After waiting for {sp.Elapsed}, could not verify that all indexes has caught up with the changes as of etag: {cutoffEtag:#,#;;0}. " +
+                               $"Total relevant indexes: {indexesToWait.Count}, total stale indexes: {staleIndexesCount}";
+
+            if (erroredIndexes.Count > 0)
```

Comment:
I do not think this is relevant anymore?

Reply:
I'm just improving the error message for the user